truffle/com.oracle.truffle.object.dsl.processor/src/com/oracle/truffle/object/dsl/processor/LayoutGenerator.java

```diff
@@ -196,7 +196,7 @@ private void generateObjectType(final PrintStream stream) {
             typeSuperclass = layout.getSuperLayout().getName() + "LayoutImpl." + layout.getSuperLayout().getName() + "Type";
         }
 
-        stream.printf("    protected static class %sType extends %s {%n", layout.getName(), typeSuperclass);
+        stream.printf("    public static class %sType extends %s {%n", layout.getName(), typeSuperclass);
 
```

Comment:
It's currently protected because other layouts need to be able to make subclass it. It is indeed visible from outside, but now that's what we want. It's restricted to the package though so it's not visible enough.